src/tools/rbd_mirror/image_sync/ObjectCopyRequest.cc

```diff
@@ -353,6 +377,7 @@ void ObjectCopyRequest<I>::compute_diffs() {
                                                          bufferlist(),
                                                          std::map<uint64_t, uint64_t>());
       }
+      m_snap_object_sizes[end_remote_snap_id] = end_size;
```

Comment:
Not sure what this is protecting against. The same end_size is encoded in the trunc op above.